### PR TITLE
feat(filter): add Filter component

### PR DESCRIPTION
**Pull request by device25**
re #252

### CODE REVIEW

**Comment by device25** [comment]
@stepankuzmin please check component interface

**Comment by stepankuzmin** [comment]
LGTM

File: src/components/Filter/README.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-
+Set filter to target layer. Layer can be define from Layer component or from mapbox style.
```
Comment (stepankuzmin) [inline src/components/Filter/README.md]: "Layer id can be an id of any layer defined in style or defined by `<Layer />` component"

File: src/components/Filter/index.js [hunk 2]
```diff
@@ -36,7 +54,11 @@ class Filter extends PureComponent<Props> {
       return;
     }
 
-    this._map.setFilter(layerId, filter);
+    if (!filter) {
```
Comment (stepankuzmin) [inline src/components/Filter/index.js]: Probably we should use `Array.isArray` here

**Comment by stepankuzmin** [comment]
Good job!

**Pull request merged**